docs/hello_nextflow/02_hello_channels.md

```diff
@@ -0,0 +1,880 @@
+
+_Before:_
+
+```groovy title="hello-channels.nf" linenums="27"
+workflow {
+
+    // emit a greeting
+    sayHello(params.greeting)
+}
+```
+
+_After:_
+
+```groovy title="hello-channels.nf" linenums="27"
+workflow {
+
+    // create a channel for inputs
+    greeting_ch = Channel.of('Hello Channels!')
+
+    // emit a greeting
+    sayHello(params.greeting)
+}
+```
+
+This is not yet functional since we haven't yet switched the input to the process call.
+
+### 1.2. Add the channel as input to the process call
+
+Now we need to actually plug our newly created channel into the `sayHello()` process call, replacing the CLI parameter which we were providing directly before.
+
+In the workflow block, make the following code change:
+
+_Before:_
+
+```groovy title="hello-channels.nf" linenums="27"
+workflow {
+
+    // create a channel for inputs
+    greeting_ch = Channel.of('Hello Channels!')
+
+    // emit a greeting
+    sayHello(params.greeting)
+}
+```
+
+_After:_
+
+```groovy title="hello-channels.nf" linenums="27"
+workflow {
+
+    // create a channel for inputs
+    greeting_ch = Channel.of('Hello Channels!')
+
+    // emit a greeting
+    sayHello(greeting_ch)
+}
+```
+
+This tells Nextflow to run the `sayHello` process on the contents of the `greeting_ch` channel.
+
+Now our workflow is properly functional; it is the explicit equivalent of writing `sayHello('Hello Channels!')`.
+
+### 1.3. Run the workflow command again
+
+Let's run it!
+
+```bash
+nextflow run hello-channels.nf
+```
+
+If you made both edits correctly, you should get another successful execution:
+
+```console title="Output" linenums="1"
+ N E X T F L O W   ~  version 24.10.0
+
+Launching `hello-channels.nf` [nice_heisenberg] DSL2 - revision: 41b4aeb7e9
+
+executor >  local (1)
+[3b/f2b109] sayHello (1) | 1 of 1 ✔
+```
+
+You can check the results directory to satisfy yourself that the outcome is still the same as previously.
+
+```console title="output.txt" linenums="1"
+Hello Channels!
+```
+
+So far we're just progressively tweaking the code to increase the flexibility of our workflow while achieving the same end result.
+
+!!! note
+
+    This may seem like we're writing more code for no tangible benefit, but the value will become clear as soon as we start handling more inputs.
+
+### Takeaway
+
+You know how to use a basic channel factory to provide an input to a process.
+
+### What's next?
+
+Learn how to use channels to make the workflow iterate over multiple input values.
+
+---
+
+## 2. Modify the workflow to run on multiple input values
+
+Workflows typically run on batches of inputs that are meant to be processed in bulk, so we want to upgrade the workflow to accept multiple input values.
+
+### 2.1. Load multiple greetings into the input channel
+
+Conveniently, the `Channel.of()` channel factory we've been using is quite happy to accept more than one value, so we don't need to modify that at all.
+We just have to load more values into the channel.
```

Comment:
I have always referred to the individual entries in a channel as elements, so as to not make people confused with what we mean by values when we talk about value channels (singletons). As we mention value channels in this part II, I'd rather call these elements. See the example below.

```
Channel.of('A', 'B', '[C, D]')
```
A, B, and [C, D] are three elements in the channel. C and D are items in the third channel.

Reply:
ok feel free to suggest changes directly; otherwise let's do this in a refinement pass after this is merged in.